From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: flange <flange@local>
Date: Mon, 13 Jul 2026 09:00:00 +0800
Subject: [PATCH] arm: dts: rk3506b: 保留 CPU2 AMP 固件内存

CPU2 的 RT-Thread 固件加载到 0x03e00000，大小 1 MiB。vendor
rk3506-amp.dtsi 只保留 SHMEM、RPMsg 和 MCU SRAM，Linux 仍会把固件区
纳入 System RAM，内存压力下可能覆盖正在运行的 RTOS。

在 ATK-RK3506B AMP 目标 DTS 中补充 no-map reserved-memory，使 Linux
页分配器排除该区域，并与 flange amp.memory/FIT load 保持一致。
---
 arch/arm/boot/dts/rk3506b-alientek-mipi720x1280-nand-ubi-ubifs-amp-linux.dts | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/boot/dts/rk3506b-alientek-mipi720x1280-nand-ubi-ubifs-amp-linux.dts b/arch/arm/boot/dts/rk3506b-alientek-mipi720x1280-nand-ubi-ubifs-amp-linux.dts
--- a/arch/arm/boot/dts/rk3506b-alientek-mipi720x1280-nand-ubi-ubifs-amp-linux.dts
+++ b/arch/arm/boot/dts/rk3506b-alientek-mipi720x1280-nand-ubi-ubifs-amp-linux.dts
@@ -22,6 +22,13 @@
         };
 };
 
+&reserved_memory {
+        amp_firmware_reserved: amp@3e00000 {
+                reg = <0x03e00000 0x00100000>;
+                no-map;
+        };
+};
+
 // UART2使用了GPIO1_C2，AMP的串口的是UART4，UART4使用了GPIO1_C2，故禁用UART2
 &uart2 {
         status = "disabled";
